pkg/pmc/pmc.go

```diff
@@ -24,50 +25,58 @@ var (
 
 // RunPMCExp ... go expect to run PMC util cmd
 func RunPMCExp(configFileName, cmdStr string, promptRE *regexp.Regexp) (result string, matches []string, err error) {
-	glog.Infof("pmc read config from /var/run/%s", configFileName)
-	glog.Infof("pmc run command: %s", cmdStr)
-	e, _, err := expect.Spawn(fmt.Sprintf("pmc -u -b 1 -f /var/run/%s", configFileName), -1)
+	pmcCmd := fmt.Sprintf("pmc -u -b 0 -f /var/run/%s", configFileName)
+	glog.Infof("%s \"%s\"", pmcCmd, cmdStr)
+	e, r, err := expect.Spawn(pmcCmd, -1)
 	if err != nil {
 		return "", []string{}, err
 	}
-	defer e.Close()
+	defer func() {
+		e.SendSignal(syscall.SIGTERM)
+		<-r
```

Comment:
need a timeout here , else it will hold the thread forever in worst case

Reply:
added timeout